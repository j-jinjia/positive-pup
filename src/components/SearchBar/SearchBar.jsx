import "./SearchBar.scss";
import searchIcon from "../../assets/svgs/searchBarIcon.svg";

const SearchBar = ({ searchTerm, handleInput }) => {
  return (
    <form className="search-bar">
      <label className="search-bar__label" htmlFor="search-bar__course">
        Search Course
      </label>
      <input
        id="search-bar__course"
        className="search-bar__input"
        type="text"
        value={searchTerm}
        onChange={handleInput}
      />
      <img
        className="search-bar__input-icon"
        src={searchIcon}
        alt="search icon"
      />
    </form>
  );
};

export default SearchBar;
