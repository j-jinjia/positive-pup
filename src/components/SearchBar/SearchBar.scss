@use "../../styles/variables" as *;

.search-bar {
  position: relative;

  &__label {
    display: none;
  }

  &__input {
    width: 15rem;
    height: 2.5rem;
    position: relative;
    border-radius: 8px;
    background-color: rgba($colour-primary, 0.04);
    border: none;

    &-icon {
      position: absolute;
      right: 10px;
      top: 10px;
    }
  }
}
