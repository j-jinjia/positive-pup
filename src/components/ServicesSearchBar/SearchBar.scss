@use "../../styles/variables" as *;


.search-box {
  position: relative;

  &__input {
    width: 15rem;
    height: 2.5rem;
    position: relative;
    border-radius: 8px;
    background-color: $colour-primary-light-searchbar;
    border: none;

    &-icon {
      position: absolute;
      right: 10px;
      top: 10px;
    }
  }
}
