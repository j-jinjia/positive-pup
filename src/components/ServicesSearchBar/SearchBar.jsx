import "./SearchBar.scss";
import searchIcon from "../../assets/svgs/searchBarIcon.svg";

const ServicesSearchBar = ({ searchTerm, handleInput }) => {
  return (
    <div className="search-box">
      <label htmlFor="search-box">
      <input
        className="search-box__input"
        type="text"
        value={searchTerm}
        onChange={handleInput}
      />
      </label>
     
      <img
        className="search-box__input-icon"
        src={searchIcon}
        alt="search icon"
      />
    </div>
  );
};

export default ServicesSearchBar;
