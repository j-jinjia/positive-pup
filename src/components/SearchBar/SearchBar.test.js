import { customRender } from "../../utils/testUtils";
import SearchBar from "./SearchBar";

it("should match SearchBar component snapshot", () => {
  const { container } = customRender(
    <SearchBar
      searchTerm="rising stars"
      label="searchbar"
      handleInput={jest.fn}
    />
  );

  expect(container).toMatchSnapshot();
});
