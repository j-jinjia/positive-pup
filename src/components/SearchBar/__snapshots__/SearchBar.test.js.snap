// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match SearchBar component snapshot 1`] = `
<div>
  <div
    class="search-bar"
  >
    <label
      class="search-bar__label"
      for="searchbar"
    >
      Searchbar
    </label>
    <input
      class="search-bar__input"
      id="searchbar"
      role="input"
      type="text"
      value="rising stars"
    />
    <img
      alt="search icon"
      class="search-bar__input-icon"
      src="searchBarIcon.svg"
    />
  </div>
</div>
`;
